spark/lib/oxygen/physicsserver/dragcontroller.cpp

```diff
@@ -54,6 +56,17 @@ void DragController::PrePhysicsUpdateInternal(float /*deltaTime*/)
         Vector3f vel = mBody->GetAngularVelocity() * mAngularDrag * -1;
         mBody->AddTorque(vel);
     }
+
+    if (mStillness > 0)
+    {
+        // apply linear drag
+        Vector3f vel = mBody->GetVelocity();
+        mBody->AddForce(vel * pow(mStillnessBase, -vel.Length()) * mStillness * mLinearDrag * -1);
+
+        // apply angular drag
+        Vector3f avel = mBody->GetAngularVelocity();
+        mBody->AddTorque(avel * pow(mStillnessBase, -avel.Length()) * mStillness * mAngularDrag * -1);
+    }
```

Comment:
Perhaps thinking about the torque, velocity and angular drag here may also lead to a better name for stillness?

Reply:
There's already a drag, so maybe dragExtra?

Reply:
Aside: There's even a [book on this](https://www.namingthings.co/) 😂 - I usually can stop at [Martin Fowler's take on it](https://www.martinfowler.com/bliki/TwoHardThings.html).

Reply:
If in doubt, I'd probably go for more descriptive, so if we can't categorise the type of `mDragExtra` as distinct from `mAngularDrag`, I'd suggest the named equivalents of `mDeceleratorBase` and `mDeceleratorFactor`

Reply:
Though as I reread the formula - any reason that `mStillness` at least (as distinct from `mStillnessBase`) can't be merged into `mAngularDrag`?

Reply:
if the built-in calculations assume a perfect sphere, and this accounts for the bumps in the ball, we could name it `mDragIrregularBase` and `mDragIrregularFactor`

Reply:
what do you mean `merged into`?  stillness applies to linear drag as well

Reply:
I'm a fan of `mDeceleratorBase` and `mDeceleratorFactor`

Perhaps we should look into making the field a little bumpy or the ball less spherical. Not sure how to atm...

Reply:
it'll make the physics calculations more complicated and the simulator will run slower.  better to add a drag in a random direction of a random amount.